zipkin-ui/js/page/trace.js

```diff
@@ -40,6 +40,18 @@ const TracePageComponent = component(function TracePage() {
                                             link: `/api/v1/trace/${this.attr.traceId}`});
       });
 
+      this.$node.find('#saveTraceLink').click(e => {
+        e.preventDefault();
+        const traceId = this.attr.traceId;
+
+        $.ajax(`/api/v1/save/trace/${traceId}`).done(result => {
```

Comment:
so basically I'd suggest having an alternative that reads from config.json and if a key like "permanentZipkinUrl" is present, show the link, and on clicking the link copy the result of /api/v2/trace/ID to a POST ${permanentZipkinEndpoint} (ex http://foo/bar/api/v2/spans) 

you don't need to process the body in any way or even know about the v2 model. The server hosting the permanent repo doesn't need to be using v2 native storage, though obviously in the case of elasticsearch it should as v2 is a better investment for a concept like permanent.

For the handling, just copy the bytes from the first response to the request body of the second. the code will be easy and the results easy to reason with. The url associated with this task could be defined in crossroads.

If this isn't clear etc, lemme know!

Reply:
one other consideration  is to add a tag to the first span (easy in v2 model), just insert one in the tags dict.
ex tags.outage='disaster happened in JIRA FOO-1234' then in your permanent store, you can search for "outage" and get this trace. Such a feature could be worth a little javascript to make it work.

This will allow you to label the trace such that in your long-term storage you can look it up somehow. Otherwise, eventually, these favorite traces will become annoying to browse.